code/modules/events/spacevine.dm

```diff
@@ -586,9 +586,14 @@
 		if(prob(mutativeness))
 			var/datum/spacevine_mutation/random_mutate = pick_weight(vine_mutations_list - vine.mutations)
```

Comment:
I don't like your current solution, consider filtering the input list first before picking a mutation.

Otherwise, this is going to result in mutations not being chosen even though the probability procs because a later 'if' check fails.

Reply:
Alternative approach with a static list so that the addresses of the mutations are always the same